packages/clay-autocomplete/src/Autocomplete.tsx

```diff
@@ -404,13 +416,50 @@ function AutocompleteInner<T extends Item>(
 		),
 		items: filteredItems,
 		notFound: (
-			<DropDown.Item
-				aria-disabled="true"
-				className="disabled"
-				roleItem="option"
-			>
-				{messages.notFound}
-			</DropDown.Item>
+			<>
+				{allowsCustomValue && (
+					<>
+						<DropDown.Item
+							className=""
+							id={value}
+							key={value}
+							onClick={() => {
+								if (allowsCustomValue && items && onAddNewItem) {
+									onAddNewItem();
+
+									items.push(value);
+
+									inputElementRef.current?.focus();
+
+									setValue('');
+								}
+							}}
+							roleItem="option"
+						>
+							{messages.setAsHTML &&
+							typeof allowsCustomValueMessage === 'string' ? (
+								<span
+									dangerouslySetInnerHTML={{
+										__html: sub(allowsCustomValueMessage, [
+											value,
+										]),
+									}}
+								/>
+							) : (
+								sub(allowsCustomValueMessage, [value])
+							)}
+						</DropDown.Item>
+						<li className="dropdown-divider"></li>
+					</>
+				)}
+				<DropDown.Item
```

Comment:
Shouldn't this only render if `allowsCustomValue` is false?

Reply:
@ethib137 I was under the impression that `allowsCustomValue === true` is for adding items not on the list.

```
	/**
	 * Flag to allow an input value not corresponding to an item to be set.
	 */
```

We could make the description better.